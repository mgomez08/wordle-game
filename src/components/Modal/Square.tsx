import { checkLetterForModal } from "../../helpers/checkLetter";
import styles from "./modal.module.scss";

interface SquareProps {
  word: string;
  solution: string;
}

export const Square = ({ word, solution }: SquareProps) => {
  return (
    <div className={styles.puzzleWord}>
      {word.split("").map((letter, index) => (
        <div key={index}>{checkLetterForModal(letter, index, solution)}</div>
      ))}
    </div>
  );
};
