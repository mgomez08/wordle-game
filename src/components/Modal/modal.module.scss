.modalViewContainer {
  width: 100%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modalContainer {
  background-color: white;
  border-radius: 3px;
  padding: 10px;
  width: 500px;
  min-height: 300px;

  h2 {
    text-align: center;
    font-size: 36px;
  }
}

.modalHeader {
  display: flex;
  justify-content: flex-end;
}

.btnClose {
  font-size: 24px;
  color: red;
}

.socialShareContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.social {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
}

.btnSocial {
  font-size: 46px;
  cursor: pointer;
}

.twitterLogo {
  color: #00aced;
}

.whatsappLogo {
  color: #25d366;
}

.puzzle {
  width: 200px;
  margin: 0 auto;
  font-size: 30px;
}

.puzzleWord {
  display: flex;
}
