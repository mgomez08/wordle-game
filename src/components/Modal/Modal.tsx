import styles from "./modal.module.scss";
import { Square } from "./Square";

interface ModalProps {
  type: "won" | "lost";
  completedWords: string[];
  solution: string;
}

export const Modal = ({ type, completedWords, solution }: ModalProps) => {
  return (
    <div className={styles.modalViewContainer}>
      <div className={styles.modalContainer}>
        <div className={styles.modalHeader}>
          <button>
            <i className={"far fa-times-circle " + styles.btnClose}></i>
          </button>
        </div>
        <div className={styles.modalContent}>
          <h2>You {type === "won" ? "Won!" : "Lost :("}</h2>
          <div className={styles.puzzle}>
            {completedWords.map((word, index) => (
              <Square key={index} word={word} solution={solution} />
            ))}
          </div>
          <div className={styles.socialShareContainer}>
            <h3>Share your score!</h3>
            <div className={styles.social}>
              <a
                className={`fab fa-twitter-square ${styles.btnSocial} ${styles.twitterLogo}`}
              ></a>
              <a
                className={`fab fa-whatsapp-square ${styles.btnSocial} ${styles.whatsappLogo}`}
              ></a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
