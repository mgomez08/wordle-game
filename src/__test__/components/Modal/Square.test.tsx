import { mount } from "enzyme";
import { Square } from "../../../components/Modal/Square";

describe("Tests for <Square />", () => {
  const square = mount(<Square word="HELLO" solution="HELLO" />);

  test("Render Square component", () => {
    expect(square.length).toEqual(1);
  });

  test("Square component should have correct class (.puzzleWord)", () => {
    expect(square.find(".puzzleWord").length).toEqual(1);
  });

  test("Square component should have correct number of children (6)", () => {
    expect(square.find("div").length).toEqual(6);
  });

  test("Square component with Snapshot", () => {
    expect(square).toMatchSnapshot();
  });
});
