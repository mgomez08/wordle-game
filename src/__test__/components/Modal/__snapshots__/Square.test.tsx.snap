// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Square /> Square component with Snapshot 1`] = `
<div
  className="puzzleWord"
>
  <div
    key="0"
  >
    游릴
  </div>
  <div
    key="1"
  >
    游릴
  </div>
  <div
    key="2"
  >
    游릴
  </div>
  <div
    key="3"
  >
    游릴
  </div>
  <div
    key="4"
  >
    游릴
  </div>
</div>
`;
